magento/pwa-studio
# [Doc] cart page reference docs

## Description

Create reference docs for the components and talons used in the cart page.
It also adds support to the docs generator for multiple child components.

## Related Issue





Closes PWA-604

## Acceptance 

Any developer

### Verification Stakeholders

Any developer

### Specification



### Verification Steps

1. Navigate to the `pwa-devdocs` directory: `cd pwa-devdocs`.
1. Build auto-generated docs: `yarn build:reference-doc-snippets`
1. Verify the following files are created under `src/_includes/auto-generated/`

- `venia-ui/lib/components/CartPage/cartPage.md`
- `venia-ui/lib/components/CartPage/GiftCards/giftCards.md`
- `venia-ui/lib/components/CartPage/PriceAdjustments/priceAdjustments.md`
- `venia-ui/lib/components/CartPage/PriceSummary/priceSummary.md`
- `venia-ui/lib/components/CartPage/ProductListing/productListing.md`
- `peregrine/lib/talons/CartPage/useCartPage.md`
- `peregrine/lib/talons/CartPage/GiftCards/useGiftCards.md`
- `peregrine/lib/talons/CartPage/GiftOptions/useGiftOptions.md`
- `peregrine/lib/talons/CartPage/PriceAdjustments/useCouponCode.md`
- `peregrine/lib/talons/CartPage/PriceSummary/usePriceSummary.md`
- `peregrine/lib/talons/CartPage/ProductListing/useProductListing.md`

4. Run the HTML preview server: `yarn develop`
1. Navigate to `Reference API -> Peregrine API` section
1. Verify new Shopping Cart Talons section in the left navigation
1. Verify links work and lead to expected topic
1. Navigate to `Reference API -> Venia UI API`
1. Verify new Shopping Cart Components section in the left navigation
1. Verify links work and lead to expected topic

## Screenshots / Screen Captures (if appropriate)

## Checklist

* I have updated the documentation accordingly, if necessary.
